engine/combat/combat_controller.py

```diff
@@ -144,34 +145,47 @@ def execute_combat(self: Self, delta: float) -> bool:
         # Check if we have control
         return False
 
-    def _handle_starting_zone(self: Self) -> None:
-        if (
-            not self.action
-            and level_manager.current_level == "72e9f2699f7c8394b93afa1d273ce67a"
-        ):
+    def _handle_encounter_controller(self: Self) -> None:
+        if not self.action:
             for player in combat_manager.players:
                 if combat_manager.selected_character == player.character:
-                    if self.second_encounter is True:
-                        match player.character:
-                            case PlayerPartyCharacter.Valere:
-                                self.action = Action(
-                                    SoSConsideration(player),
-                                    CrescentArc(timing_type=SoSTimingType.NONE),
-                                )
-                            case PlayerPartyCharacter.Zale:
-                                self.action = Action(
-                                    SoSConsideration(player),
-                                    Sunball(value=1000, hold_time=2.0),
-                                )
-                    else:
-                        match player.character:
-                            case PlayerPartyCharacter.Valere:
-                                self.action = Action(
-                                    SoSConsideration(player),
-                                    BasicAttack(timing_type=SoSTimingType.NONE),
-                                )
-                            case PlayerPartyCharacter.Zale:
-                                self.action = Action(
-                                    SoSConsideration(player),
-                                    BasicAttack(timing_type=SoSTimingType.NONE),
-                                )
+                    match combat_manager.combat_controller:
+                        case CombatEncounter.FirstEncounter:
+                            match player.character:
+                                case PlayerPartyCharacter.Valere:
+                                    self.action = Action(
+                                        SoSConsideration(player),
+                                        BasicAttack(timing_type=SoSTimingType.NONE),
+                                    )
+                                case PlayerPartyCharacter.Zale:
+                                    self.action = Action(
+                                        SoSConsideration(player),
+                                        BasicAttack(timing_type=SoSTimingType.NONE),
+                                    )
+
+                        case CombatEncounter.SecondEncounter:
+                            if self.second_encounter is True:
```

Comment:
`second_encounter` here is rather second turn, of said second encounter?

Reply:
renamed, but this will make sense more more when these encounters are extracted to custom logic.